src/esockd_transport.erl

```diff
@@ -330,6 +330,16 @@ peer_cert_common_name(#ssl_socket{ssl = SslSock}) ->
 peer_cert_common_name(#proxy_socket{pp2_additional_info = AdditionalInfo}) ->
     proplists:get_value(pp2_ssl_cn,
                         proplists:get_value(pp2_ssl, AdditionalInfo, [])).
+-spec(peersni(socket()) -> undefined | binary()).
+peersni(Sock) when is_port(Sock) ->
+    undefined;
+peersni(#ssl_socket{ssl = SslSock}) ->
+    case ssl:connection_information(SslSock, [sni_hostname]) of
+        {ok, [{sni_hostname, SNI}]} -> iolist_to_binary(SNI);
```

Comment:
according to the doc, the type spec is as: `{sni_hostname, [term()](https://www.erlang.org/doc/man/erlang.html#type-term)}`

not sure when it will return a non-iolist `term()`, maybe `undefined` ? 
but maybe better to wrap in a try catch.